app/controllers/homes_controller.rb

```diff
@@ -1,4 +1,95 @@
   def index
+    def consumer_key
+      OAuth::Consumer.new(
+      ENV["TWITTER_CONSUMER_KEY"],
+      ENV["TWITTER_CONSUMER_SECRET"])
+    end
+
+    def access_token
+      OAuth::Token.new(
+      ENV["TWITTER_ACCESS_TOKEN"],
+      ENV["TWITTER_ACCESS_SECRET"])
+    end
+
+    def api_get(url)
+      address = URI(url)
+      request = Net::HTTP::Get.new address.request_uri
+      http = Net::HTTP.new address.host, address.port
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      request.oauth! http, consumer_key, access_token
+      http.start
+      response = http.request request
+      response
+    end
+
+    client = Twitter::REST::Client.new do |config|
+      config.consumer_key        = ENV["TWITTER_CONSUMER_KEY"]
+      config.consumer_secret     = ENV["TWITTER_CONSUMER_SECRET"]
+      config.access_token        = ENV["TWITTER_ACCESS_TOKEN"]
+      config.access_token_secret = ENV["TWITTER_ACCESS_SECRET"]
+    end
+
+    ip_address = request.remote_ip unless Rails.env.test? || Rails.env.development?
+    ip_address = "50.241.127.209" if Rails.env.test? || Rails.env.development?
+    location = GeoIP.new('GeoLiteCity.dat').city(ip_address)
+    @lat = location.latitude
+    @lng = location.longitude
+
+    @tweets = []
+    @local_trends = []
+    q = "geocode:39.5,-98.35,1500mi"
+    all_tweets = client.search(q).take(50)
+    if all_tweets != nil
+      @tweets = all_tweets.delete_if {|t| /(#|)(job|hiring|work)/i.match(t.text)}
+
+      raw = []
+      @tweets.each do |t|
+        words = t.text.split
+        words.each do |w|
+          raw << w
+        end
+      end
+      common_words = raw.find_all { |e| raw.count(e) > 2 && e.length > 3 }.uniq!
+      @local_trends = common_words.delete_if { |w| /(have|this|with|just|your|when|&amp;|from|that|-&gt;|were)/i.match(w)}
+    end
+
+    available_trends_response = api_get("https://api.twitter.com/1.1/trends/available.json")
+    available_trends_info = nil
+    if available_trends_response.code == '200'
+      available_trends_info = JSON.parse(available_trends_response.body)
+      us_trends = available_trends_info.select { |i| i["countryCode"] == "US" }[0..5]
+    end
+
+    trend_locations = []
+    us_trends.each do |trend|
+      trend_locations << trend["woeid"]
+    end
+
+    us_trend_markers = Hash.new {|h, k| h[k] = ''}
+    trend_locations.each do |place|
+      trend_location_response = api_get("https://api.twitter.com/1.1/trends/place.json?id=#{place}")
+      trend_location_info = nil
+      if trend_location_response.code == '200'
+        trend_location_info = JSON.parse(trend_location_response.body)
+        trend_name = trend_location_info.first["trends"][0]["name"]
+        us_trend_markers.store(trend_name, place)
+      end
+    end
+
+    @remote_trends = []
+    us_trend_markers.each do |trend, place|
+      location = HTTParty.get("http://where.yahooapis.com/v1/place/#{place}?format=json&appid=#{ENV["YAHOO"]}")
```

Comment:
Line is too long. [111/80]<br>Prefer single-quoted strings inside interpolations.